### PR TITLE
MULE-18984: Dynamic Configurations are not expired until the flow ends.

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java [hunk 1]
```diff
@@ -85,6 +118,28 @@ public void expirationWorksAfterRestartingSource() throws Exception {
     doNotExpireDynamicConfigWithCustomExpirationUsedBySource();
   }
 
+  @Test
+  public void dynamicConfigIsExpiredBeforeFlowEnds() throws Exception {
+    flowRunner("dynamicConfigIsExpiredBeforeFlowEnds").withPayload("Walter Blanco").run();
+    check(PROBER_TIMEOUT, POLLING_FREQUENCY, () -> {
+      synchronized (disposedStatuses) {
+        return disposedStatuses.size() == 2;
```
[the remark this answers is not in the document]
Comment (SebaElizalde) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java:126]: What do you mean? Can you ellaborate?

[the remark this answers is not in the document]
Comment (marianogonzalez) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java:126]: the condition should be `>=` but asciidoc attacked

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java [hunk 2]
```diff
@@ -66,6 +66,13 @@ public void setUp() {
     config = null;
   }
 
+  @Override
+  protected void doTearDown() throws Exception {
+    disposedStatuses = new ArrayList<>();
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java:71]: null

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Comment by SebaElizalde** [comment]
--validate

**Pull request merged**